lib/pure/random.nim

```diff
@@ -279,7 +277,11 @@ proc rand*[T: Ordinal or SomeFloat](max: T): T {.benign.} =
 
     let f = rand(1.0)
     ## f = 8.717181376738381e-07
```

Comment:
pre-existing but how about the more informative: `doAssert x >= 0.0 and x <= 1.0`